### PR TITLE
Fix `extract_masked_elements` unflatten bug

### PR DESCRIPTION
**Pull request by jc-audet**
**What does this PR do? Please describe:**

The function `extract_masked_elements` in `models/wav2vec2/masker.py` is not behaving as expected because it is breaking down the pipeline on my setup and the current behavior of `seqs` does not seems to align with the commented shapes.

Specifically the N dimension in the unflatten operation is referring to the `size(0)` of a (N x T, M) tensor which leads to a tensor of shape (N x T, 1, M), while the intended output is (N, T, M). This PR fixes the problem on my setup.

**Does your PR introduce any breaking changes? If yes, please list them:**
List of all backwards-incompatible changes.

**Check list:**
- [x] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [x] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**?
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

File: src/fairseq2/models/wav2vec2/masker.py [hunk 1]
```diff
@@ -149,8 +149,11 @@ def extract_masked_elements(seqs: Tensor, temporal_mask: Tensor) -> Tensor:
         The temporal mask. *Shape:* :math:`(N,S)`, where :math:`N` is the batch
         size and :math`S` is the sequence length.
     """
+    
+    N=seqs.size(0)
```
Comment (cbalioglu) [inline src/fairseq2/models/wav2vec2/masker.py:153]: Nit: do you mind naming this variable `batch_size`? I know it is more verbose, but, as a convention, we use that variable name in several places in the code base.

**Review by cbalioglu — COMMENTED** [review]
Just left a nit comment. And looks like black complains about formatting issues. Let me know once you address them, I can stamp and merge it right away.

File: src/fairseq2/models/wav2vec2/masker.py (hunk 1 above)
Comment (jc-audet) [reply src/fairseq2/models/wav2vec2/masker.py]: Done!

**Review by cbalioglu — APPROVED** [review]
LGTM!

**Pull request merged**